test/test_stackprof.rb

```diff
@@ -126,19 +126,33 @@ def test_custom
   end
 
   def test_raw
+    before_monotonic = Process.clock_gettime(Process::CLOCK_MONOTONIC, :microsecond)
+
     profile = StackProf.run(mode: :custom, raw: true) do
       10.times do
         StackProf.sample
       end
     end
 
+    after_monotonic = Process.clock_gettime(Process::CLOCK_MONOTONIC, :microsecond)
+
     raw = profile[:raw]
     assert_equal 10, raw[-1]
     assert_equal raw[0] + 2, raw.size
 
     offset = RUBY_VERSION >= '3' ? -3 : -2
     assert_includes profile[:frames][raw[offset]][:name], 'StackProfTest#test_raw'
+
+    assert_equal 10, profile[:raw_sample_timestamps].size
+    profile[:raw_sample_timestamps].each_cons(2) do |t1, t2|
+      assert_operator t1, :>, before_monotonic
+      assert_operator t2, :>=, t1
+      assert_operator t2, :<, after_monotonic
+    end
+
     assert_equal 10, profile[:raw_timestamp_deltas].size
+    total_duration = after_monotonic - before_monotonic
+    assert_operator profile[:raw_timestamp_deltas].sum, :<, total_duration
```

Comment:
I wanted to also assert >0 but FWIW on both this branch and master I actually see negative time deltas on my machine (an M1 Macbook) in this test case. I assume this is only possible because we're sampling in a tight loop, have never seen it in prod, but it's weird and I couldn't figure out from the code how it could happen. I should probably file a separate bug for that.